app/src/main/java/com/android/bookswap/data/source/network/BookFirestoreSource.kt

```diff
@@ -151,10 +156,21 @@ class BooksFirestoreSource(private val db: FirebaseFirestore) : BooksRepository
     Log.d("BooksFirestoreRepository", "Attempting to add book: ${dataBook.title}")
 
     // Attempt to add book to Firestore
-    val bookDocument = bookToDocument(dataBook)
+    val bookMap =
```

Comment:
The usage of the bookToDocument function is better and should probably be maintained here, as it can be used further in other functions. Given that a DataBook is passed as a parameter and used for the map, and given that the DataBook now consists of all previous variables with the addition of the two new ones, I cannot understand why you did not maintain the usage of bookToDocument; this is especially true given that you modified bookToDocument in order to account for these new variables. If this is due to a misunderstanding on my part, please excuse it and hopefully given an explanation I may understand the absence of those two variables being mapped.